openassessment/xblock/test/test_staff_area.py

```diff
@@ -1291,6 +1339,31 @@ def _setup_xblock_and_create_submission(self, xblock, anonymous_user_id='Bob', t
         """
         A shortcut method to setup ORA xblock and add a user submission or a team submission to the block.
         """
+        self._setup_xblock(xblock, anonymous_user_id=anonymous_user_id, team=team, has_team=has_team)
+        if team:
+            arbitrary_test_user = UserFactory.create()
+            return self._create_team_submission(
+                STUDENT_ITEM['course_id'],
+                xblock.location,
+                MOCK_TEAM_ID,
+                arbitrary_test_user.id,
+                xblock.get_anonymous_user_ids_for_team(),
+                {'parts': [{'text': 'This is a team response'}]}
+            )
+        else:
+            student_item = STUDENT_ITEM.copy()
+            student_item["item_id"] = xblock.location
+            return self._create_submission(student_item, {
+                'text': "Text Answer",
+                'file_keys': kwargs.get('file_keys', []),
+                'files_descriptions': kwargs.get('files_descriptions', []),
+                'files_names': kwargs.get('files_names', [])
+            }, ['staff'])
+
+    def _setup_xblock(self, xblock, anonymous_user_id='Bob', team=False, has_team=True):
```

Comment:
I feel like this logic, or at least a good chunk of it, should maybe live in a share-able location for other modules, as setting up the teams service appropriately is v time and code-consuming.